src/lib.rs

```diff
@@ -46,50 +50,92 @@ impl Clipboard {
         Self { request_receiver, request_sender, clipboard_thread }
     }
 
-    /// Load clipboard data.
-    ///
-    /// Loads content from a clipboard on a last observed seat.
-    pub fn load(&self) -> Result<String> {
-        let _ = self.request_sender.send(worker::Command::Load);
+    fn load_inner<T: AllowedMimeTypes + 'static>(&self, target: SelectionTarget) -> Result<T>
+    where
+        <T as TryFrom<(Vec<u8>, MimeType)>>::Error: std::error::Error + Send + Sync,
+    {
+        let _ = self.request_sender.send(worker::Command::Load(T::allowed().to_vec(), target));
 
         if let Ok(reply) = self.request_receiver.recv() {
-            reply
+            match reply {
+                Ok((data, mime)) => T::try_from((data, mime))
+                    .map_err(|err| std::io::Error::new(std::io::ErrorKind::Other, err)),
+                Err(err) => Err(err),
+            }
         } else {
             // The clipboard thread is dead, however we shouldn't crash downstream, so
             // propogating an error.
             Err(std::io::Error::new(std::io::ErrorKind::Other, "clipboard is dead."))
         }
     }
```

Comment:
Can probably rewrite this with `match` + `.recv().map`, it's just ugly the way it's right now.